server/server_test.go

```diff
@@ -1,94 +1,148 @@
+	// The MongoSuite from IE provides useful features for starting/stopping
+	// a test mongo database, as well as inserting test fixtures.
+	testutil.MongoSuite
+	PTMergeServer *httptest.Server
+	FHIRServer    *httptest.Server
+	patientIDs    []string
 }
 
 func TestServerTestSuite(t *testing.T) {
 	suite.Run(t, new(ServerTestSuite))
 }
 
 func (s *ServerTestSuite) SetupSuite() {
-	// set gin to release mode (less verbose output)
+	// Set gin to release mode (less verbose output).
 	gin.SetMode(gin.ReleaseMode)
 
-	// build routes for testing
-	s.Engine = gin.New()
-	RegisterRoutes(s.Engine)
+	// Create a mock FHIR server to check the ptmerge service's outgoing requests. The first
+	// call to s.DB() stands up the mock Mongo server, see testutil/mongo_suite.go for more.
+	fhirEngine := gin.New()
+	ms := server.NewMasterSession(s.DB().Session, "ptmerge-test")
+	server.RegisterRoutes(fhirEngine, nil, server.NewMongoDataAccessLayer(ms, nil), server.Config{})
+	s.FHIRServer = httptest.NewServer(fhirEngine)
+
+	// POST test fixtures (bundles) to the mock FHIR server.
+	s.patientIDs = append(s.patientIDs, s.postPatientBundle("../fixtures/john_peters_bundle.json"))
+	s.patientIDs = append(s.patientIDs, s.postPatientBundle("../fixtures/clint_abbot_bundle.json"))
+
+	// Create a mock PTMergeServer.
+	ptmergeEngine := gin.New()
+	RegisterRoutes(ptmergeEngine, s.DB().Session, s.FHIRServer.URL)
+	s.PTMergeServer = httptest.NewServer(ptmergeEngine)
+}
+
+// Inserts a fixture into the mock FHIR server.
+func (s *ServerTestSuite) postPatientBundle(fixturePath string) string {
+	// Load and POST the fixture.
+	data, err := os.Open(fixturePath)
+	s.NoError(err)
+	defer data.Close()
+	res, err := http.Post(s.FHIRServer.URL+"/", "application/json", data)
+	s.NoError(err)
+	defer res.Body.Close()
 
-	// create HTTP test server
-	s.Server = httptest.NewServer(s.Engine)
+	// Check the response.
+	s.Equal(http.StatusOK, res.StatusCode)
+	decoder := json.NewDecoder(res.Body)
+	responseBundle := &models.Bundle{}
+	err = decoder.Decode(responseBundle)
+	s.NoError(err)
+	s.Equal("transaction-response", responseBundle.Type)
+
+	// Return the new ID of the Patient resource.
+	patient, ok := responseBundle.Entry[0].Resource.(*models.Patient)
+	s.True(ok)
+	s.Equal("Patient", patient.Resource.ResourceType)
+	return patient.Resource.Id
 }
 
 func (s *ServerTestSuite) TeardownSuite() {
-	s.Server.Close()
+	s.PTMergeServer.Close()
+	s.FHIRServer.Close()
+	// Clean up and remove all temporary files from the mocked database.
+	// See testutil/mongo_suite.go for more.
+	s.TearDownDBServer()
 }
 
 func (s *ServerTestSuite) TestInitiateMerge() {
-	source1 := "http://www.example.com/fhir/Patient/12345"
-	source2 := "http://www.example.com/fhir/Patient/67890"
-	req := s.Server.URL + "/merge?source1=" + url.QueryEscape(source1) + "&source2=" + url.QueryEscape(source2)
+	source1 := s.FHIRServer.URL + "/Patient?_id=" + s.patientIDs[0] + "&_include=*&_revinclude=*"
+	source2 := source1
```

Comment:
Fair point. I just used the same URL since the `Merger` is only stubbed out right now. In practice I plan to add additional tests not just for a valid merge, but also for all the corner cases where things could go wrong.